lib/neovim-plugin.nix

```diff
@@ -157,7 +166,55 @@
                 ])
                 ++ (lib.optionals hasConfigAttrs [
                   (lib.optionalAttrs callSetup { ${namespace}.${name}.luaConfig.content = setupCode; })
-                  (lib.optionalAttrs (configLocation != null) (setLuaConfig cfg.luaConfig.content))
+                  (lib.mkIf (!cfg.lazyLoad.enable) (
+                    lib.optionalAttrs (configLocation != null) (setLuaConfig cfg.luaConfig.content)
+                  ))
+                ])
+                ++ (lib.optionals hasConfigAttrs [
+                  (lib.mkIf (config.plugins.lz-n.enable && cfg.lazyLoad.enable && cfg.lazyLoad.backendSettings != { })
+                    {
+                      plugins.lz-n = {
+                        plugins = [
+                          {
+                            __unkeyed-1 = originalName;
+                            # Use provided after, otherwise fallback to normal lua content
+                            after =
+                              if cfg.lazyLoad.backendSettings.after != null then
+                                cfg.lazyLoad.backendSettings.after
+                              else
+                                # We need to wrap it in a function so it doesn't execute immediately
+                                "function()\n " + cfg.luaConfig.content + " \nend";
+                            inherit (cfg.lazyLoad.backendSettings)
+                              enabled
+                              priority
+                              before
+                              beforeAll
+                              event
+                              cmd
+                              ft
+                              keys
+                              colorscheme
+                              ;
+                            extraSettings = lib.filterAttrs (
+                              name: value:
+                              !(lib.elem name [
+                                "enabled"
+                                "priority"
+                                "before"
+                                "beforeAll"
+                                "event"
+                                "cmd"
+                                "ft"
+                                "keys"
+                                "colorscheme"
+                                "after"
+                              ])
+                            ) cfg.lazyLoad.backendSettings;
```

Comment:
1. This can use `removeAttrs`
2. Does lz-n not have a freeform settings option?